RedHatInsights/tasks-frontend
# feat(RHINENG-8942): Connect RunTaskModal to a separate route

Implements https://issues.redhat.com/browse/RHINENG-8942.

This implements a route component for RunTaskModal that allows to render a modal and tie it to a separate URL route from the available tasks page. Other views (like executed list, executed task details) are not affected. Routes are also refactored in order to accommodate the change.

## How to test

1. Go to available task (home page)
2. Open any task from the list and make sure the URL is updated
3. Refresh the page and validate that the modal is opened automatically
4. In the pathname, try to add any random slug of non-existent task and make sure the error message is shown in a modal
